src/controllers/books.js

```diff
@@ -34,14 +38,52 @@ exports.getBorrowedBooks = (request, response) => {
     });
 };
 
-exports.getAddBookTab = (request, response) => {
-  response.render('view_books',
-    {
-      addBookTab: 'active',
-      book: 'active',
-      layout: 'admin',
-      title: 'الكتب',
-      style: 'book',
-      js: 'book',
+exports.getAddBookTab = (request, response, next) => {
+  getCategory()
+    .then((res) => {
+      response.render('view_books',
+        {
+          res,
+          addBookTab: 'active',
+          book: 'active',
+          layout: 'admin',
+          title: 'الكتب',
+          style: 'book',
+          js: 'book',
+        });
+    }).catch((err) => {
+      next(err);
+    });
+};
+
+exports.addCategory = (req, response) => {
+  const categoryData = req.body;
+  setCategory(categoryData)
+    .then((results) => {
+      const result = { errorMessage: null, message: 'category Added !' };
+      response.send(JSON.stringify(result));
+    })
+    .catch((err) => {
+      const errorMessage = err.detail;
+      if (err.code === '23505') {
+        const result = { errorMessage: 'This category is Already Exists' };
+        return response.send(JSON.stringify(result));
+      }
+      const result = { errorMessage };
+      return response.send(JSON.stringify(result));
```

Comment:
Is this server error? use `response.status(500)`
same with line`70`

Reply:
line 70 I want send error the message from back-end to front-end and alert the message to admin
this category already exists but can replace line 73 response.status(500) or next(err)

Reply:
@RamyAlshurafa

Reply:
I think if the request sent by fetch you should use `response.status(500)`
but if you use the form to send the request use `next(err)`